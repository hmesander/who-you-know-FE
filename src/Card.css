
.card-component {
  border-top: 1px solid #eee;
  background-color: #fff;
  box-shadow:
    0 2px 2px rgba(0,0,0,0.15),
    0 11px 0 -5px #eee,
    0 11px 1px -4px rgba(0,0,0,0.15),
    0 22px 0 -10px #eee,
    0 22px 1px -9px rgba(0,0,0,0.15);
  width: 75%;
  min-height: fit-content;
  height: 60vh;
  margin: 40px auto;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  padding: 30px;
}

.circle-buttons {
  border-radius: 50%;
  margin: 36px;
  width: 65px;
  height: 65px;
  text-align: center;
  font-weight: 800;
  position: relative;
  top: 50px;
  box-shadow: 0px 3px 8px #aaa, inset 0px 2px 3px #fff;
  background-image: linear-gradient(to left, #f7f7f7 0%,#e7e7e7 100%);
}

.circle-image {
  border-radius: 50%;
  height: 150px;
  width: 150px;
}

.card-hr {
  width: 90%;
}

.card-button-container {
  /* margin-bottom: 50px; */
}