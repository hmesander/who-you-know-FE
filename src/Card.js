import React from 'react';
import './Card.css';

const Card = (props) => {
  return (
    <div className='card-component'>
      <div className='card-info-section'>
        <img src="https://pbs.twimg.com/profile_images/512675211309633536/lYHh4p7W_400x400.jpeg" alt="Smiley face" className="circle-image"></img>
        <h3 className=''>Chet Manley</h3>
        <h6>ISIS Field Agent</h6>
        <h6>Georgetown University</h6>
        <hr className='card-hr'></hr>
      </div>
      <div className='card-button-container'>
        <button className='circle-buttons'>No Clue</button>
        <button className='circle-buttons'>Vague Idea</button>
        <button className='circle-buttons'>BFF</button>
      </div>
    </div>
  );
};

export default Card;