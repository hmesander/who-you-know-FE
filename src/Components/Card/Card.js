import React from 'react';
import './Card.css';

const Card = (props) => {
  
  if (props.cardsToSort) {
    return (
      <div className='card-component'>
        <div className='card-info-section'>
          <img src={props.cardsToSort[0].imgUrl} alt="Smiley face" className="circle-image"></img>
          <h3 className=''>{props.cardsToSort[0].name}</h3>
          <h6>{props.cardsToSort[0].title}</h6>
          <h6>{props.cardsToSort[0].school}</h6>
          <hr className='card-hr'></hr>
        </div>
        <div className='card-button-container'>
          <button 
            className='circle-buttons'
            onClick={() => { props.handleUpVote() }}
          >No Clue</button>
          <button 
            className='circle-buttons'
            onClick={() => { props.handleMiddleVote() }}
          >Vague Idea</button>
          <button 
            className='circle-buttons' 
            onClick={() => { props.handleUpVote() }}>BFF</button>
        </div>
      </div>
    );
  } else if (props.cardsToPlay && props.cardsToPlay.length) {
    return (
      <div className='card-component'>
        <div className='card-info-section'>
          <img src={props.cardsToPlay[0].imgUrl} alt="Smiley face" className="circle-image"></img>
          <h3 className=''>{props.cardsToPlay[0].name}</h3>
          <hr className='card-hr'></hr>
        </div>
        <div className='card-button-container'>
          <button
            className='circle-buttons'
          >Neither</button>
          <button
            className='circle-buttons'
          >School</button>
          <button
            className='circle-buttons'
            >Job Title</button>
        </div>
      </div>
    );
  } else {
    return (
      <div>
        No cards to sort
      </div>
    );
  }

 
};

export default Card;