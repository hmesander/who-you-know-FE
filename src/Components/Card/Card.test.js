/*eslint-disable no-undef, camelcase*/

import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

describe('CARD TESTS', () => {
  it('should match the snapshot with cards to sort', () => {
    const mockCards = [{first_last_name: 'John Doe', job_title: 'Developer', school: 'CU'}];
    const wrapper = shallow(<Card cardsToSort={mockCards} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot without cards to sort', () => {
    const mockCards = [];
    const wrapper = shallow(<Card cardsToSort={mockCards} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleVote when the upvote button is clicked', () => {
    const mockVote = jest.fn();
    const mockCards = [{ first_last_name: 'John Doe', job_title: 'Developer', school: 'CU' }];
    const wrapper = shallow(<Card cardsToSort={mockCards} handleVote={mockVote}/>);

    const button = wrapper.find('.upvote');
    button.simulate('click');
    expect(mockVote).toHaveBeenCalled();  
  });

  it('should call handleVote when the middle vote button is clicked', () => {
    const mockVote = jest.fn();
    const mockCards = [{ first_last_name: 'John Doe', job_title: 'Developer', school: 'CU' }];
    const wrapper = shallow(<Card cardsToSort={mockCards} handleVote={mockVote} />);

    const button = wrapper.find('.middlevote');
    button.simulate('click');
    expect(mockVote).toHaveBeenCalled();
  });

  it('should call handleVote when the down vote button is clicked', () => {
    const mockVote = jest.fn();
    const mockCards = [{ first_last_name: 'John Doe', job_title: 'Developer', school: 'CU' }];
    const wrapper = shallow(<Card cardsToSort={mockCards} handleVote={mockVote} />);

    const button = wrapper.find('.downvote');
    button.simulate('click');
    expect(mockVote).toHaveBeenCalled();
  });
});