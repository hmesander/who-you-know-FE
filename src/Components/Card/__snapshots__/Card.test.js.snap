// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CARD TESTS should match the snapshot with cards to sort 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardsToSort={
        Array [
            Object {
              "first_last_name": "John Doe",
              "job_title": "Developer",
              "school": "CU",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-info-section"
>
          <img
                    alt="Profile"
                    className="circle-image"
                    src={undefined}
          />
          <h3
                    className=""
          >
                    John Doe
          </h3>
          <h6>
                    Developer
          </h6>
          <h6>
                    CU
          </h6>
          <hr
                    className="card-hr"
          />
</div>,
        <div
          className="card-button-container"
>
          <button
                    className="circle-buttons downvote"
                    onClick={[Function]}
          >
                    No Clue
          </button>
          <button
                    className="circle-buttons middlevote"
                    onClick={[Function]}
          >
                    Vague Idea
          </button>
          <button
                    className="circle-buttons upvote"
                    onClick={[Function]}
          >
                    BFF
          </button>
</div>,
      ],
      "className": "card-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Profile"
              className="circle-image"
              src={undefined}
/>,
            <h3
              className=""
>
              John Doe
</h3>,
            <h6>
              Developer
</h6>,
            <h6>
              CU
</h6>,
            <hr
              className="card-hr"
/>,
          ],
          "className": "card-info-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Profile",
              "className": "circle-image",
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "John Doe",
              "className": "",
            },
            "ref": null,
            "rendered": "John Doe",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Developer",
            },
            "ref": null,
            "rendered": "Developer",
            "type": "h6",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "CU",
            },
            "ref": null,
            "rendered": "CU",
            "type": "h6",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "card-hr",
            },
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="circle-buttons downvote"
              onClick={[Function]}
>
              No Clue
</button>,
            <button
              className="circle-buttons middlevote"
              onClick={[Function]}
>
              Vague Idea
</button>,
            <button
              className="circle-buttons upvote"
              onClick={[Function]}
>
              BFF
</button>,
          ],
          "className": "card-button-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No Clue",
              "className": "circle-buttons downvote",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "No Clue",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Vague Idea",
              "className": "circle-buttons middlevote",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Vague Idea",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "BFF",
              "className": "circle-buttons upvote",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "BFF",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-info-section"
>
            <img
                        alt="Profile"
                        className="circle-image"
                        src={undefined}
            />
            <h3
                        className=""
            >
                        John Doe
            </h3>
            <h6>
                        Developer
            </h6>
            <h6>
                        CU
            </h6>
            <hr
                        className="card-hr"
            />
</div>,
          <div
            className="card-button-container"
>
            <button
                        className="circle-buttons downvote"
                        onClick={[Function]}
            >
                        No Clue
            </button>
            <button
                        className="circle-buttons middlevote"
                        onClick={[Function]}
            >
                        Vague Idea
            </button>
            <button
                        className="circle-buttons upvote"
                        onClick={[Function]}
            >
                        BFF
            </button>
</div>,
        ],
        "className": "card-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Profile"
                className="circle-image"
                src={undefined}
/>,
              <h3
                className=""
>
                John Doe
</h3>,
              <h6>
                Developer
</h6>,
              <h6>
                CU
</h6>,
              <hr
                className="card-hr"
/>,
            ],
            "className": "card-info-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Profile",
                "className": "circle-image",
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "John Doe",
                "className": "",
              },
              "ref": null,
              "rendered": "John Doe",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Developer",
              },
              "ref": null,
              "rendered": "Developer",
              "type": "h6",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CU",
              },
              "ref": null,
              "rendered": "CU",
              "type": "h6",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "card-hr",
              },
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="circle-buttons downvote"
                onClick={[Function]}
>
                No Clue
</button>,
              <button
                className="circle-buttons middlevote"
                onClick={[Function]}
>
                Vague Idea
</button>,
              <button
                className="circle-buttons upvote"
                onClick={[Function]}
>
                BFF
</button>,
            ],
            "className": "card-button-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No Clue",
                "className": "circle-buttons downvote",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "No Clue",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Vague Idea",
                "className": "circle-buttons middlevote",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Vague Idea",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "BFF",
                "className": "circle-buttons upvote",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "BFF",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`CARD TESTS should match the snapshot without cards to sort 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardsToSort={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          No cards to sort.
</p>,
        <button
          className="play-deck-button buttons"
          onClick={undefined}
>
          Play a Deck
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No cards to sort.",
        },
        "ref": null,
        "rendered": "No cards to sort.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Play a Deck",
          "className": "play-deck-button buttons",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Play a Deck",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            No cards to sort.
</p>,
          <button
            className="play-deck-button buttons"
            onClick={undefined}
>
            Play a Deck
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No cards to sort.",
          },
          "ref": null,
          "rendered": "No cards to sort.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Play a Deck",
            "className": "play-deck-button buttons",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Play a Deck",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
